packages/nitro-protocol/contracts/ninja-nitro/SingleChannelAdjudicator.sol

```diff
@@ -481,4 +452,17 @@ contract SingleChannelAdjudicator is
             return IERC20(assetHolderAddress).balanceOf(address(this));
         }
     }
+
+    function _holdings(address channelAddress, address assetHolderAddress)
+        internal
+        view
+        returns (uint256)
+    {
+        if (assetHolderAddress == address(0)) {
+            return channelAddress.balance;
+        } else {
+            // assume ERC20 Token
```

Comment:
There are many token standards: see for example [ERC777](https://docs.openzeppelin.com/contracts/3.x/erc777 ) . This one is backwards compatible with ERC20, so we would already support it out of the box. 

ERC721 tokens are *non-fungible*, which is problematic for state channel networks. It's not inconceivable that a directly funded channel might hold an NFT -- cross-chain swaps of NFTs is one potential use case. But you couldn't virtually fund such a channel, or otherwise send it across multiple hops, because those constructions require several "copies" of the asset to be locked up by different parties.